### PR TITLE
ROBOT-20098 : Added Socker client to execute glusterfs command on Vmexec

### PR DESCRIPTION
**Pull request by venjetti**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by wdec — APPROVED** [review]

File: kadalu_operator/main.py [hunk 1]
```diff
@@ -466,7 +466,7 @@ def handle_external_storage_addition(core_v1_client, obj):
     """Deploy service(One service per Volume)"""
     volname = obj["metadata"]["name"]
     details = obj["spec"]["details"]
-    pv_reclaim_policy = obj["spec"].get("pvReclaimPolicy", "delete")
+    pv_reclaim_policy = obj["spec"].get("pvReclaimPolicy", "retain")
```
Comment (wdec) [inline kadalu_operator/main.py]: Why change the default? The less kadalu core changes are done, the better.

File: csi/volumeutils.py [hunk 2]
```diff
@@ -639,7 +639,7 @@ def delete_volume(volname):
     with open(os.path.join(VOLINFO_DIR, storage_filename)) as info_file:
         storage_data = json.load(info_file)
 
-    pv_reclaim_policy = storage_data.get("pvReclaimPolicy", "delete")
+    pv_reclaim_policy = storage_data.get("pvReclaimPolicy", "retain")
```
Comment (wdec) [inline csi/volumeutils.py]: Why?

**Review by wdec — CHANGES_REQUESTED** [review]
Guys, this PR needs changes:
1. Revert the changes to the base kadalu py code in terms of the default retention getter return value.
2. SQUASH the commits. We shouldn't have 20 commits!

File: kadalu_operator/main.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def validate_volume_request(obj):
         logging.error("Storage 'spec' not specified")
         return False
 
-    pv_reclaim_policy = obj["spec"].get("pvReclaimPolicy", "delete")
+    pv_reclaim_policy = obj["spec"].get("pvReclaimPolicy", "retain")
```
Comment (wdec) [inline kadalu_operator/main.py]: All of these .py changes should be reverted

File: csi/volumeutils.py (hunk 2 above)
Comment (venjetti) [reply csi/volumeutils.py]: Will do this now.

File: manifests/kadalu-operator-openshift.yaml [hunk 4]
```diff
@@ -37,7 +37,7 @@ spec:
                   type: string
                 pvReclaimPolicy:
                   type: string
-                  default: delete
+                  default: retain
```
Comment (wdec) [inline manifests/kadalu-operator-openshift.yaml]: And also revert the ones here for templates that we're not using

File: csi/volumeutils.py [hunk 5]
```diff
@@ -1080,15 +1087,17 @@ def mount_glusterfs_with_host(volname, mountpoint, hosts, options=None, is_clien
...
-             ))
+        logging.info(logf(
+                "proceeding with xlator",
+                 cmd=cmd,
+                 ))
 
     for host in hosts.split(','):
-        cmd.extend(["--volfile-server", host+":24007"])
+        if netaddr.valid_ipv6(host):
+            host += ":24007"
```
Comment (wdec) [inline csi/volumeutils.py]: Are you sure about this? An IPv6 address can next to never be extended with a port :X. The IPv6 numeral typically needs to have square braces

**Review by wdec — CHANGES_REQUESTED** [review]

File: csi/volumeutils.py (hunk 5 above)
Comment (venjetti) [reply csi/volumeutils.py]: We reverted this port value that is added previously for ipv6 case. Now we see some errors in ipv6, so testing having this change back.

File: manifests/kadalu-operator-openshift.yaml (hunk 4 above)
Comment (venjetti) [reply manifests/kadalu-operator-openshift.yaml]: Will check with Rajesh if these are required and revert these changes. Thanks.

Comment (wdec) [reply manifests/kadalu-operator-openshift.yaml]: These are not needed. We don't use openshift, etc. Revert changes for all the templates we don't use and keep the ones only for the template that we do.

File: csi/volumeutils.py (hunk 5 above)
Comment (wdec) [reply csi/volumeutils.py]: Reverted where?

**Pull request closed without merging**